coordinator/points_writer.go

```diff
@@ -121,6 +121,15 @@ func (s *ShardMapping) MapPoint(shardInfo *meta.ShardInfo, p models.Point) {
 	s.Shards[shardInfo.ID] = shardInfo
 }
 
+func withinWriteWindow(rp *meta.RetentionPolicyInfo, p models.Point) bool {
+	if (rp != nil) &&
+		(((rp.FutureWriteLimit > 0) && p.Time().After(time.Now().Add(rp.FutureWriteLimit))) ||
+			((rp.PastWriteLimit > 0) && p.Time().Before(time.Now().Add(-rp.PastWriteLimit)))) {
```

Comment:
Suggestion for future optimization: Create a `writeWindow` object that caches the value of `time.Now().Add(rp.FutureWriteLimit)` and `time.Now().Add(-rp.PastWriteLimit)` to avoid calculating them twice for every point in request.